supabase/migrations/20_auto_discovers.sql

```diff
@@ -0,0 +1,112 @@
+
+
+create view internal.next_auto_discovers as
+select
+  live_specs.id as capture_id,
+  live_specs.catalog_name as capture_name,
+  live_specs.spec->'endpoint' as endpoint_json,
+  -- These properties default to false, which matches the behavior in the models crate.
+  coalesce((live_specs.spec->'autoDiscover'->>'addNewBindings')::boolean, false) as add_new_bindings,
+  coalesce((live_specs.spec->'autoDiscover'->>'evolveIncompatibleCollections')::boolean, false) as evolve_incompatible_collections,
+  connector_tags.id as connector_tags_id,
+  -- If there's not been any discovers, then we use the capture creation time as the starting point, so that we don't auto-discover
+  -- immediately after a capture is created. This is also required in order to effectively disable auto-discover by setting the
+  -- auto_discover_interval to a really large value. Note that this expression must be consistent with the 'having' clause.
+  now() - coalesce(max(discovers.updated_at), live_specs.created_at) + connector_tags.auto_discover_interval as overdue_interval
+from live_specs
+left join discovers on live_specs.catalog_name = discovers.capture_name
+-- We can only perform discovers if we have the connectors and tags rows present.
+-- I'd consider it an improvement if we could somehow refactor this to log a warning in cases where there's no connector_tag
+inner join connectors
+  on split_part(live_specs.spec->'endpoint'->'connector'->>'image', ':', 1) = connectors.image_name
+inner join connector_tags
+  on connectors.id = connector_tags.connector_id
+  and ':' || split_part(live_specs.spec->'endpoint'->'connector'->>'image', ':', 2) = connector_tags.image_tag
+where
+  live_specs.spec_type = 'capture'
+  -- We don't want to discover if shards are disabled
+  and not coalesce((live_specs.spec->'shards'->>'disabled')::boolean, false)
+  -- Any non-null value for autoDiscover will enable it.
+  and live_specs.spec->'autoDiscover' is not null
```

Comment:
We'll initiate the discover and only try to update existing bindings. We'll still try to publish it, since it's possible that there's schema changes that just don't cause anything to break. The meaning of `evolveIncompatibleCollections` is that an evolutions job should be created _if_ that publication fails. Otherwise, we'll want to trigger some sort of alert, which is not yet a capability we have.